pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -78,7 +91,11 @@ def main(argv=None):  # type: (Optional[Sequence[str]]) -> int
     for filename in args.filenames:
         _, extension = os.path.splitext(filename.lower())
         md = all_markdown or extension in md_exts
-        if _fix_file(filename, md):
+        if _fix_file(
```

Comment:
this line is doing a bit much, maybe instead:

```python
chars = args.chars.encode('UTF-8') if args.chars else None

...

for filename in args.filenames:

    ...

    if _fix_file(filename, md, chars):
        ...
```
```

(this also lifts the argument parsing out of the loop instead of doing it N times)

Reply:
Addressed in commit a33a8f0.